### PR TITLE
add villagers_follow_emerald_blocks

### PR DESCRIPTION
**Pull request by SFort**
> #83 (Features/Todo)
unascribed/Fabrication-Features#51

### CODE REVIEW

File: src/main/java/com/unascribed/fabrication/mixin/g_weird_tweaks/villagers_follow_emerald_blocks/MixinVillagerEntity.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import java.util.stream.StreamSupport;
+
+@Mixin(VillagerEntity.class)
+@EligibleIf(configEnabled="*.villagers_follow_emerald_blocks")
+public abstract class MixinVillagerEntity extends MerchantEntity  {
+	PlayerEntity player = null;
```
Comment (exaskye) [inline src/main/java/com/unascribed/fabrication/mixin/g_weird_tweaks/villagers_follow_emerald_blocks/MixinVillagerEntity.java]: This name is too generic, and the field should be `private`.

```suggestion
	private PlayerEntity fabrication$player = null;
```

File: src/main/java/com/unascribed/fabrication/mixin/g_weird_tweaks/villagers_follow_emerald_blocks/MixinVillagerEntity.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	private static final TargetPredicate TARGET_PREDICATE =
+			(new TargetPredicate()).setBaseMaxDistance(10.0).includeInvulnerable().includeTeammates().ignoreDistanceScalingFactor().ignoreEntityTargetRules().setPredicate(
+					player -> StreamSupport.stream(player.getItemsHand().spliterator(), false).anyMatch(
```
Comment (exaskye) [inline src/main/java/com/unascribed/fabrication/mixin/g_weird_tweaks/villagers_follow_emerald_blocks/MixinVillagerEntity.java]: I really don't like the usage of streams here; they're incredibly inefficient. Does vanilla do it this way?

File: src/main/java/com/unascribed/fabrication/mixin/g_weird_tweaks/villagers_follow_emerald_blocks/MixinVillagerEntity.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import java.util.stream.StreamSupport;
+
+@Mixin(VillagerEntity.class)
+@EligibleIf(configEnabled="*.villagers_follow_emerald_blocks")
+public abstract class MixinVillagerEntity extends MerchantEntity  {
+	PlayerEntity player = null;
+	private static final TargetPredicate TARGET_PREDICATE =
```
Comment (exaskye) [inline src/main/java/com/unascribed/fabrication/mixin/g_weird_tweaks/villagers_follow_emerald_blocks/MixinVillagerEntity.java]: Name is too generic again.
```suggestion
	private static final TargetPredicate FABRICATION$TARGET_PREDICATE =
```

**Review by exaskye — CHANGES_REQUESTED** [review]

File: src/main/java/com/unascribed/fabrication/mixin/g_weird_tweaks/villagers_follow_emerald_blocks/MixinVillagerEntity.java (hunk 2 above)
Comment (SFort) [reply src/main/java/com/unascribed/fabrication/mixin/g_weird_tweaks/villagers_follow_emerald_blocks/MixinVillagerEntity.java]: I'll take a look and change it to whatever vanilla does. (didn't know they are inefficent, they resembled rust iterators so i just assumed their better) and the reason i used iteration at all was for some reason i though: what if there's a mod that adds a hand.

**Review by exaskye — APPROVED** [review]

**Pull request merged**